standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java

```diff
@@ -253,7 +253,8 @@ public String toString() {
       ConfVars.PARTITION_NAME_WHITELIST_PATTERN,
       ConfVars.CAPABILITY_CHECK,
       ConfVars.DISALLOW_INCOMPATIBLE_COL_TYPE_CHANGES,
-      ConfVars.EXPRESSION_PROXY_CLASS
+      ConfVars.EXPRESSION_PROXY_CLASS,
+      ConfVars.METASTORE_METADATA_TRANSFORMER_CLASS
```

Comment:
As discussed above https://github.com/apache/hive/pull/4341#discussion_r1213915718, we just **put external only when the transformer is null, else let it go its own way**.  Therefore, i need to check the value of transformer property `METASTORE_METADATA_TRANSFORMER_CLASS`, as this property is only for metastore conf(hivemetastore-site.xml) used by HMSHandler, i can not get its real value in` iceberg-handler` package.

I added the property in here, so that i can get its value by `HiveMetastoreClient::getMetConf`, like this:
https://github.com/apache/hive/pull/4341/files#diff-7c252c3ca08c9b9604de9899cf2637d9ea9fb5134eadc6c8549fa612c0f1b48dR252

I also think this may expose some risk  that use may change it using hms client by accident. But I haven't thought of any other better way to achieve this.
Could you give me some suggestion. Thanks.

Reply:
Can pass something as part of environment context, create_table_as_external and post the metadata transformer is done processing, if the table doesn't have external keyword we can put external in TBLPROPERTIES
```

    if (transformer != null) {
      tbl = transformer.transformCreateTable(tbl, processorCapabilities, processorId);
    }

Post this check:-
    
    // check if table isn't exteral & has env context set
    // -> If yes, and has create_always_as_external set to true then add external in TBLPROPERTIES

```